src/giskard_hub/client.py

```diff
@@ -63,34 +65,52 @@ def __init__(
         api_key : str, optional
             API key to authenticate with the Giskard Hub. If not provided, it
             will be read from the `GSK_API_KEY` env variable.
+        auto_add_api_suffix : bool, optional
+            If True, automatically adds the `_api` suffix to the `hub_url` if not already present.
 
         Raises
         ------
         ValueError
             If the `hub_url` or `api_key` are not provided and the environment
             variables are not set.
+        HubConnectionError
+            If calling the `/openapi.json` endpoint fails (e.g. bad URL or server error).
         """
         if hub_url is None:
             hub_url = os.getenv("GSK_HUB_URL")
         if hub_url is None:
             raise ValueError(
                 "Missing Giskard Hub URL. Please provide it as an argument or set the env variable `GSK_HUB_URL`"
             )
+        hub_url = hub_url.rstrip("/")
+        if "/_api" not in hub_url and auto_add_api_suffix:
+            hub_url += "/_api"
 
         if api_key is None:
             api_key = os.getenv("GSK_API_KEY")
+
         if api_key is None:
             raise ValueError(
                 "Missing Giskard Hub API key. Please provide it as an argument or with the env variable `GSK_API_KEY`"
             )
-        self._hub_url = hub_url.rstrip("/")
+
+        self._hub_url = hub_url
         self._api_key = api_key
 
         super().__init__(**kwargs)
 
         # Set base url on the client
         self._http.base_url = self._hub_url
 
+        # Check if the connection is valid
+        try:
+            resp = self._http.get("/openapi.json")
+            resp.raise_for_status()
+        except Exception as e:
+            raise HubConnectionError(
+                f"Failed to connect to Giskard Hub at {self._hub_url}. Error: {e}"
```

Comment:
I don't think  'the Error: {e}' is bringing much info, since there is the 'from e' part, since it's in the stack